starknet-rpc-test/call.rs

```diff
@@ -0,0 +1,232 @@
+    );
+
+    Ok(())
+}
+
+#[rstest]
+#[tokio::test]
+async fn fail_incorrect_calldata(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
+    let madara = madara.await;
+    let rpc = madara.get_starknet_client();
+
+    madara.create_empty_block().await?;
+
+    assert_matches!(
+        rpc.call(
+            FunctionCall {
+                contract_address: FieldElement::from_hex_be(FEE_TOKEN_ADDRESS).unwrap(),
+                entry_point_selector: get_selector_from_name("name").unwrap(),
+                calldata: vec![FieldElement::ONE] // name function has no calldata
+            },
+            BlockId::Tag(BlockTag::Latest)
+        )
+        .await
+        .err(),
+        Some(ProviderError::StarknetError(StarknetErrorWithMessage {
+            message: _,
+            code: MaybeUnknownErrorCode::Known(StarknetError::ContractError)
+        }))
+    );
+
+    Ok(())
+}
+
+#[rstest]
+#[tokio::test]
+async fn works_on_correct_call_no_calldata(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
+    let madara = madara.await;
+    let rpc = madara.get_starknet_client();
+
+    madara.create_empty_block().await?;
+
+    assert_eq!(
+        rpc.call(
+            FunctionCall {
+                contract_address: FieldElement::from_hex_be(FEE_TOKEN_ADDRESS).unwrap(),
+                entry_point_selector: get_selector_from_name("name").unwrap(),
+                calldata: vec![] // name function has no calldata
+            },
+            BlockId::Tag(BlockTag::Latest)
+        )
+        .await
+        .unwrap(),
+        vec![FieldElement::ZERO]
+    );
+
+    Ok(())
+}
+
+#[rstest]
+#[tokio::test]
+async fn works_on_correct_call_with_calldata(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
+    let madara = madara.await;
+    let rpc = madara.get_starknet_client();
+
+    madara.create_empty_block().await?;
+
+    assert!(
+        rpc.call(
+            FunctionCall {
+                contract_address: FieldElement::from_hex_be(FEE_TOKEN_ADDRESS).unwrap(),
+                entry_point_selector: get_selector_from_name("balanceOf").unwrap(),
+                calldata: vec![FieldElement::TWO] // name function has no calldata
+            },
+            BlockId::Tag(BlockTag::Latest)
+        )
+        .await
+        .unwrap()[0]
+            .gt(&FieldElement::ZERO)
+    );
+
+    Ok(())
+}
+
+#[rstest]
+#[tokio::test]
+async fn works_on_mutable_call_without_modifying_storage(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
+    let madara = madara.await;
+    let rpc = madara.get_starknet_client();
+
+    madara.create_empty_block().await?;
+    let account = create_account(rpc, SIGNER_PRIVATE, ARGENT_CONTRACT_ADDRESS);
+
+    let contract_artifact: SierraClass = serde_json::from_reader(
+        std::fs::File::open(env!("CARGO_MANIFEST_DIR").to_owned() + "/contracts/HelloStarknet.sierra.json").unwrap(),
+    )
+    .unwrap();
+
+    let declaration = account.declare(
```

Comment:
can you rename it to `declare_tx` or any name that make it clear it is a transaction

Reply:
fixed